File: stacks.cpp
#include <iostream>
#include <string>

using namespace std ;

#define MAX 5


class Stack
{
    private:
    int values[MAX];
    int top;
    
    public:
    Stack()
    {
        top = -1;
    }

    //to push
    void push(int num)
    {
        if (top == MAX -1)
        {
            cout << "Stack Overflow. \n";
            return;
    
        }
        top++;
        values[top]= num;
        
        cout <<"Pushed" << num << "to stack." << endl ;
    }
    void pop()
    {
        if (top == -1)
        {
            cout<< "Stack Underflow. \n";
            return;
        }
        cout << values[top] << "removed. \n";
        top --;
        
    }
    
    void peek()
    {
        if (top == -1)
        {
            cout << "nothing to see. \n";
            return;
        }
        cout << "top value is:  " << values[top] << endl;
    }
    void display()
    {
        if (top == -1)
        {
            cout << "nothing to see. \n";
            return;
        }
        cout << "Values in Stack: \n";
        for (int i=top; i>=0; i--)
        {
            cout << values[i] << endl;
        }
    }
};

int main()
{
    Stack s;
    int choice;
    int num;
    char ans;
    
    do
    {
        cout << "---- Stack Menu ----" <<endl;
        cout << "1. Push" <<endl;
        cout << "2. Pop" <<endl;
        cout << "3. Peek" <<endl;
        cout << "4. Display" <<endl;
        cout << "Enter your choice :" <<endl;
        cin >> choice;

        switch(choice)
        {
            case 1:
            cout <<"Enter Value:    " << endl;
            cin >> num;
            s.push(num);
            break;

            case 2:
            s.pop();
            break;

            case 3:
            s.peek();
            break;

            case 4:
            s.display();
            break;

            default :
            cout <<"Invalid, Try again!" <<endl;

        }
        cout<<"Do you want to continue?(y/n):"<<endl;
        cin>>ans;
    } 
    while (ans == 'Y' || ans == 'y');
    return 0;
    
};

File: Stacks.py
#making a fixed size stack in python

class fixedstack:
    def __init__(self, capacity: int):
        self.capacity = capacity
        self.stack = [None]* capacity
        self.top = -1

    def isempty(self) -> bool:
        return self.top == -1

    def isfull(self) -> bool:
        return self.top == self.capacity -1

    #push
    def push(self, item):
        if self.isfull():
            print ("Stack overflow!")

        else:
            self.top += 1 #increment the top pointer
            self.stack[self.top] = item #change the top to now be inserted element

    #pop
    def pop(self):
        if self.isempty():
            print ("Stack Underflow!")

        else:
            item = self.stack[self.top] #assign the top item to some other variable/placeholder
            self.stack[self.top] = None #change the top to none
            self.top -= 1 #decrement the top pointer
            return item #return the previous top item which is now popped

    def peek(self):
        if self.isempty():
            print ("Stack is empty!")

        else:
            print ("Top element is  :   ", self.stack[self.top])

    def display(self):
        if self.isempty():
            print ("Stack is empty!")

        else:
            active_elements = self.stack [:self.top +1] #sliced to remove "none"/empty slots
            print ("your Stack  :")
            for i in reversed(active_elements): #puts the list in reverse to get bottom to top print vertically
                print (i)

if  __name__ == "__main__":
    capacity = int(input("Enter stack capacity  :"))
    stack = fixedstack(capacity)

    while True:
        print ("----Stack Menu----")
        print ("1.Push \n2.Pop \n3.Peek \n4.Display \n5.Exit")

        choice = int(input("Enter your choice   :"))

        if choice == 1:
            item = (input("Enter element to push    :"))
            stack.push(item)

        elif choice == 2:
            popped = stack.pop()

            print ("Item popped :", popped)

        elif choice == 3:
            stack.peek()

        elif choice == 4:
            stack.display()

        elif choice == 5:
            print("Byee!!")
            break

        else:
            print("Invalid Try again!")
